### PR TITLE
Doc Update Formats

### PR DESCRIPTION
**Pull request by smk78**
This PR addresses https://github.com/SasView/sasview/issues/1921

Work so far:

testdata_help has been completely overhauled (requiring the creation of a new label in each of fitting_help and pr_help).
data_formats_help has been completely overhauled
Added a hyperlink to marketplace.rst for good measure.

### CODE REVIEW

File: src/sas/qtgui/MainWindow/media/data_formats_help.rst [hunk 1]
```diff
@@ -10,91 +10,188 @@
...
+          legacy formats to be converted into either the canSAS SASXML format
+          or the NeXus NXcanSAS format. These legacy formats include BSL/OTOKO,
+          output from FIT2D and some other SAXS-oriented software, and the ISIS
+          COLETTE (or 'RKH') 2D format.
 
 1D SAS Formats
 --------------
 
-SasView recognizes 1D data in ASCII, CSV, CanSAS XML, NXCanSAS, and AntonPaar SAXSESS formats with the following extensions:
+SasView recognizes 1D data either arranged in separated columns in ASCII
```
[the remark this answers is not in the document]
Comment (krzywon) [reply src/sas/qtgui/MainWindow/media/data_formats_help.rst:28]: @butlerpd sasmodels assumes the supplied dQ is 1-sigma of a guassian for pinhole resolution.

[the remark this answers is not in the document]
Comment (butlerpd) [reply src/sas/qtgui/MainWindow/media/data_formats_help.rst:28]: really?!!! 1-sigma? not just sigma?  That seems wrong. Are you sure?! 1/2sigma even I would understand.  This definitely needs documenting!

[the remark this answers is not in the document]
Comment (krzywon) [reply src/sas/qtgui/MainWindow/media/data_formats_help.rst:28]: From [sasmodels.resolution.py](https://github.com/SasView/sasmodels/blob/master/sasmodels/resolution.py), line 67: 

> *q_width* gaussian 1-sigma resolution at each data point.

[the remark this answers is not in the document]
Comment (smk78) [reply src/sas/qtgui/MainWindow/media/data_formats_help.rst:28]: Are we sure that doesn't just mean _one_ sigma?

[the remark this answers is not in the document]
Comment (butlerpd) [reply src/sas/qtgui/MainWindow/media/data_formats_help.rst:28]: That would make more sense.   sigma could in principle be more than 1 (not in our Q space of course but in general principle .. right?)

[the remark this answers is not in the document]
Comment (krzywon) [reply src/sas/qtgui/MainWindow/media/data_formats_help.rst:28]: Yes, one sigma, not one minus sigma. And looking at the pinhole smearing calculation, that is what it is used as.

[the remark this answers is not in the document]
Comment (butlerpd) [reply src/sas/qtgui/MainWindow/media/data_formats_help.rst:28]: Excellent.  That will be good to document here. I was under the impression that it was FWHM for example

**Comment by smk78** [comment]
@butlerpd , @krzywon : Have just pushed a further revision based on our extended exchange and @krzywon 's detective work.

**Comment by butlerpd** [comment]
ready for testing on Win

**Comment by dehoni** [comment]
> 
> 
> I think this should work though looking at the NIST IGOR wiki site I see there are an array of format (including a GRASP type output) which are not necessarily supported by SasView ... Is this too confusing? If so we could point to the two formats that are definitely supported. If not then what is done is simpler and more likely to not change url names with time? 😃
> 
> I will try to build the documentation just to test the links and the rendering before approving I guess if that is OK .. but happy to merge now as well if there is an urgency.

As a sidenote for the NIST people, looking at the NCNR output documentation for [Grasp](https://github.com/sansigormacros/ncnrsansigormacros/wiki/NCNROutput2D_GRASP), this seems to be a rather old format for 2D data, that is not used anymore in Grasp. Currently 2D data are stored in a .nxs format (not loadable in SASview, somehow deviating from the expectation) and a text file that is similiar to the NIST 2D ASCII format (just missing ASCII data as header), that is usable for SASview.

File: src/sas/qtgui/MainWindow/media/data_formats_help.rst [hunk 2]
```diff
@@ -10,91 +10,188 @@
...
+
+**Examples of this format can be found in the \\test\\sesans_data sub-folder
+in the SasView installation folder.**
+
+.. ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ
+
+Coordinate Formats
+------------------
+
+The :ref:`SANS_Calculator_Tool` in SasView recognises ASCII ('text') files
```
[the remark this answers is not in the document]
Comment (dehoni) [reply src/sas/qtgui/MainWindow/media/data_formats_help.rst:133]: The calculator recognises Ascii data assuming rectangular pixels (grid type data) like OMF and SLD. The PDB format is read as grid type data with a pixel volumes to tweak the atomic volume [specification here](https://www.wwpdb.org/documentation/file-format).
The new vtk format would also  allow other regular elements like tetrahedra or hexahedra for finite elements type data, but this is part of the open PR [ticket-1820-polygonal-fourier](https://github.com/SasView/sasview/tree/ticket-1820-polygonal-fourier).

[the remark this answers is not in the document]
Comment (smk78) [reply src/sas/qtgui/MainWindow/media/data_formats_help.rst:139]: Thanks @dehoni . I have incorporated this feedback in commit https://github.com/SasView/sasview/pull/1933/commits/4202eb5c97b46daa55c80c876d31f9826066f722. And tweaked the 2D format text a little.

**Comment by smk78** [comment]
Addressed review feedback by @dehoni above.

**Review by krzywon — APPROVED** [review]
After a reread, I think everything is correct now, but merging might depend on @smk78's comment:

> TO DO: if we merge PR #1918 will it be necessary to add mention of VTK files, and examples?

**Comment by dehoni** [comment]
The VTK files shouldn't hold back the PR. When ticket 1918 is merged adding a link to the section in the General scattering calculator is sufficient where the supported structure is explained in further detail.

**Comment by smk78** [comment]
As data_formats.rst said that VTK was _not_ supported, I've just pushed an edit to remove that in anticipation of https://github.com/SasView/sasview/issues/1918 being merged. If time permits before the release a further edit could be pushed to link to the updated General Scattering Calculator docs.

**Pull request merged**